internal/descriptor/provider/provider.go

```diff
@@ -70,6 +71,20 @@ func (c *CachedDescriptorProvider) GetDescriptor(template *v1beta2.ModuleTemplat
 	return descriptor, nil
 }
 
+func (c *CachedDescriptorProvider) SetDescriptor(template *v1beta2.ModuleTemplate, descriptor *v1beta2.Descriptor,
+) error {
+	if template == nil {
+		return ErrTemplateNil
+	}
+	template.Spec.Descriptor.Object = descriptor
+	encoded, err := compdesc.Encode(descriptor.ComponentDescriptor, compdesc.DefaultJSONCodec)
+	template.Spec.Descriptor.Raw = encoded
+	if err != nil {
+		return errors.Join(ErrEncode, err)
+	}
+	return nil
+}
```

Comment:
This is production code that is only used for testing. Such things should be avoided in general (sometimes not possible though). Here is a proposal for how we can fix the integration test without adding new production code:

https://github.com/LeelaChacha/kyma-lifecycle-manager/pull/1

Reply:
that is a really clean solution! The only downside I see is the extra file, but objectively better than extra functions only for test.